web/src/app/workplace/taskdetails/taskdetails.component.ts

```diff
@@ -11,24 +11,37 @@ import {Location} from '@angular/common';
 })
 export class TaskdetailsComponent implements OnInit {
   task: Task = null;
+  requestInProgress = false;
+
+  private routeSubscription: Subscription;
 
   constructor(private route: ActivatedRoute,
               private taskService: TaskService,
-              private location: Location) {
+              private router: Router) {
   }
 
   ngOnInit() {
-    this.getTask();
+    this.routeSubscription = this.route.params.subscribe(params => {
+      this.requestInProgress = true;
```

Comment:
why exactly is this better? Should it not be the same?

Reply:
Because requestInProgress is for async calls, if you follow the flow of this two functions you will find the next execution order, since taskService.getTask(id)  is returning an observable, so You are setting requestInProgress to false before tasks are being retrieved.

this.routeSubscription = this.route.params.subscribe(params => {
[1]
    this.requestInProgress = true;
     const id = params['id'];
     this.getTask(id);
[3]
      this.requestInProgress = false;
   });

getTask(id: string): void {
[2]
     this.taskService.getTask(id).subscribe(task => {
[4]
    this.task = task;   });
 }

------------------ correct way

getTask(id: string): void {
this.requestInProgress = true;
     this.taskService.getTask(id).subscribe(task => {
     this.requestInProgress = false;
     this.task = task;   });
 }